internal/store/common.go

```diff
@@ -297,3 +299,158 @@ func retryUpdate(fn func() (bool, error)) error {
 	}
 	return err
 }
+
+func configsAreEqual(c1, c2 *[]api.DeviceSpec_Config_Item) bool {
+	if c1 == nil {
+		return c2 == nil
+	}
+	if c2 == nil {
+		return false // c2 is nil, c1 is not
+	}
+
+	if len(*c1) != len(*c2) {
```

Comment:
The first 2 conditions can be skipped and use only this one

Reply:
Fixed.